android/app/src/main/java/org/pathcheck/covidsafepaths/MainActivity.java

```diff
@@ -91,126 +62,11 @@ private void checkIfExposureNotificationsEnabled() {
         );
   }
 
-  public void showPermissionShareKeys(ApiException apiException) {
-    try {
-      apiException
-          .getStatus()
-          .startResolutionForResult(
-              this, RequestCodes.REQUEST_CODE_GET_TEMP_EXPOSURE_KEY_HISTORY);
-    } catch (IntentSender.SendIntentException e) {
-      getExposureKeysPromise.reject(e);
-    }
-  }
-
   @Override
   public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
     super.onActivityResult(requestCode, resultCode, data);
-    onResolutionComplete(requestCode, resultCode);
-  }
-
-  /**
-   * Called when opt-in resolution is completed by user.
-   *
-   * <p>Modeled after {@code Activity#onActivityResult} as that's how the API sends callback to
-   * apps.
-   */
-  public void onResolutionComplete(int requestCode, int resultCode) {
-    if (requestCode == RequestCodes.REQUEST_CODE_START_EXPOSURE_NOTIFICATION) {
-      ReactContext context = getReactContext();
-      ExposureNotificationClientWrapper exposureNotificationClient =
-          ExposureNotificationClientWrapper.get(context);
-      if (resultCode == Activity.RESULT_OK) {
-        exposureNotificationClient.start(context);
-      } else {
-        exposureNotificationClient.onExposureNotificationStateChanged(context, false);
-      }
-    } else if (requestCode == RequestCodes.REQUEST_CODE_GET_TEMP_EXPOSURE_KEY_HISTORY) {
-      if (resultCode == RESULT_OK) {
-        getExposureKeys(getExposureKeysPromise);
-      } else {
-        // Don't share.
-        getExposureKeysPromise.reject("CANCEL", "Operation cancelled by the user");
-      }
-    }
-  }
-
-  public void getExposureKeys(final Promise promise) {
-    getExposureKeysPromise = promise;
-    FluentFuture<ImmutableList<DiagnosisKey>> getKeys =
-        FluentFuture.from(getRecentKeys())
-            .transform(
-                this::toDiagnosisKeysWithTransmissionRisk, AppExecutors.getLightweightExecutor());
-
-    Futures.addCallback(
-        getKeys,
-        new FutureCallback<ImmutableList<DiagnosisKey>>() {
-          @Override
-          public void onSuccess(ImmutableList<DiagnosisKey> shared) {
-            final List<RNExposureKey> exposureKeys = new ArrayList<>();
-
-            for (DiagnosisKey k : shared) {
-              final String key = BASE64.encode(k.getKeyBytes());
-              final int rollingStartNumber = k.getIntervalNumber();
-
-              exposureKeys.add(new RNExposureKey(
-                  key,
-                  DEFAULT_PERIOD,
```

Comment:
We were sending to JS this DEFAULT_PERIOD (144) for all the keys returned by the API.
I'm now using the `rollingPeriod` that is present in the `TemporaryExposureKey` object returned by the API.

@aledustet Is this the right thing to do, or should I keep using the default value?

Reply:
Here it says it's 144 https://blog.google/documents/69/Exposure_Notification_-_Cryptography_Specification_v1.2.1.pdf not sure where else we could look for that value